src/client/Microsoft.Identity.Client/ManagedIdentity/V2/IHybridCache.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.ManagedIdentity.V2
+{
+    /// <summary>
+    /// Interface for attestation token caching implementations.
+    /// Supports hybrid caching strategies combining in-memory and persistent storage
+    /// for attestation tokens used in managed identity authentication scenarios.
+    /// </summary>
+    /// <remarks>
+    /// This interface abstracts the caching mechanism to allow for different storage strategies:
+    /// - Hybrid caching with in-memory primary and file-based fallback for optimal performance
+    /// - Pure in-memory caching for single-process scenarios
+    /// - Pure persistent file-based caching for cross-process scenarios
+    /// - Custom implementations for specific requirements
+    /// 
+    /// Implementations should be thread-safe and handle concurrent access gracefully.
+    /// For persistent implementations, cross-process synchronization should be considered.
+    /// Hybrid implementations should synchronize between memory and persistent storage when possible.
+    /// </remarks>
+    internal interface IHybridCache
```

Comment:
I am not sure about the name, as it conflicts with https://learn.microsoft.com/en-us/aspnet/core/performance/caching/hybrid?view=aspnetcore-9.0